### PR TITLE
Add an opts to allow non-pluralized types

### PR DESCRIPTION
**Pull request by digia**

### CODE REVIEW

File: test/serializer.js [hunk 1]
```diff
@@ -30,6 +30,37 @@ describe('Options', function () {
     });
   });
 
+  describe('pluralizeType', function () {
+    it('should allow type to not be pluralized', function (done) {
+      var dataSet = {
+        id: '1',
+        firstName: 'Sandro',
+        lastName: 'Munda',
+      }
```
Comment (SeyZ) [inline test/serializer.js]: missing `;` here :)

File: test/serializer.js [hunk 2]
```diff
@@ -30,6 +30,37 @@ describe('Options', function () {
...
+
+      new JsonApiSerializer('user', dataSet, {
+        attributes: ['firstName', 'lastName'],
+        pluralizeType: false
+      }).then(function (json) {
+        console.log(json);
```
Comment (SeyZ) [inline test/serializer.js]: Can you just remove the console.log ?

File: test/serializer.js [hunk 3]
```diff
@@ -30,6 +30,37 @@ describe('Options', function () {
...
+        expect(json.data.type).equal('user');
+
+        // Confirm it response the same with a truthy setting
+        new JsonApiSerializer('user', dataSet, {
+          attributes: ['firstName', 'lastName'],
+          pluralizeType: true
+        }).then(function (json) {
+          console.log(json);
```
Comment (SeyZ) [inline test/serializer.js]: Same here, can you just remove the `console.log` ?

File: test/serializer.js [hunk 4]
```diff
@@ -30,6 +30,37 @@ describe('Options', function () {
...
+          expect(json.data.type).equal('users');
+          done(null, json);
+        });
+
```
Comment (SeyZ) [inline test/serializer.js]: Can you remove this blank line?

File: test/serializer.js [hunk 5]
```diff
@@ -30,6 +30,37 @@ describe('Options', function () {
...
+      });
+
```
Comment (SeyZ) [inline test/serializer.js]: Can you remove this blank line?

File: test/serializer.js [hunk 6]
```diff
@@ -30,6 +30,37 @@ describe('Options', function () {
...
+    });
+
```
Comment (SeyZ) [inline test/serializer.js]: Can you remove this blank line?

File: lib/serializer-utils.js [hunk 7]
```diff
@@ -21,6 +21,9 @@ module.exports = function (collectionName, payload, opts) {
   }
 
   function getType(str) {
+    if ( ! _.isUndefined(opts.pluralizeType) && ! opts.pluralizeType) {
+      return str;
+    }
     return inflection.pluralize(str);
   }
```
Comment (SeyZ) [inline lib/serializer-utils.js]: What about configuring the default value of the pluralizeType when the serialization utils is instantiated?
Example:

```
module.exports = function (collectionName, payload, opts) {
  // Default value: true
  var pluralizeType = _.isUndefined(opts.pluralizeType) || opts.pluralizeType;
  ...
  ...

  function getType(str) {
    return pluralizeType ? inflection.pluralize(str) : str;
  }
```

**Comment by SeyZ** [comment]
@digia Thank you very much for your contribution :+1:

**Comment by digia** [comment]
@SeyZ No problem! And apologize for the mess. That's what i get for rushing a push before running out the door.

File: lib/serializer-utils.js (hunk 7 above)
Comment (digia) [inline lib/serializer-utils.js]: That would work as well. Is that how you want it?

**Comment by SeyZ** [comment]
Good job @digia :+1:

**Comment by SeyZ** [comment]
I squashed your commits and now, it's merge ;)

**Pull request closed without merging**